TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceResource.robot

```diff
@@ -0,0 +1,130 @@
+*** Settings ***
+Resource     TAF/testCaseModules/keywords/common/commonKeywords.robot
+Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
+Suite Setup  Run Keywords  Setup Suite
+...                        AND  Run Keyword if  $SECURITY_SERVICE_NEEDED == 'true'  Get Token
+Suite Teardown  Run Teardown Keywords
+Force Tags      v2-api
+
+*** Variables ***
+${SUITE}          Core Metadata Device Profile DELETE Device Resource Positive Test Cases
+${LOG_FILE_PATH}  ${WORK_DIR}/TAF/testArtifacts/logs/core-metadata-deviceprofile-delete-deviceresource-positive.log
+
+*** Test Cases ***
+ProfileResourceDELETE001 - Delete deviceResource by name
+    Given Set Test Variable  ${profile_name}  Test-Profile-1
+    And Generate A Device Profile Sample  ${profile_name}
+    And Create device profile ${deviceProfile}
+    When Delete Unused deviceResource with by Name
+    Then Should Return Status Code "200"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    And Device Resource ${resource_name} Should Be Deleted
+    [Teardown]  Delete device profile by name  ${profile_name}
+
+ErrProfileResourceDELETE001 - Delete deviceResource by non-existent deviceResource name
```

Comment:
Please add a testcase for non-existent device profile.

Reply:
Updated.